File: src/Application Specific/Classes/Entity Collections/Jobs/JobOfWork/JobOfWorkValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace JobOfWorks
    {
        // make sure that contact object is valid
        public class JobOfWorkValidator : BaseValidator
        {
            public void Validate(JobOfWork oJobOfWork)
            {

                // make sure that contact object is valid
                if (oJobOfWork.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oJobOfWork.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Jobs/JobOfWork/JobOfWorkValidator.vb
Namespace Entity

Namespace JobOfWorks

Public Class JobOfWorkValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oJobOfWork As JobOfWork)

        'make sure that contact object is valid
        If oJobOfWork.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oJobOfWork.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace
